vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/PollOptionViewStateFactory.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.home.room.detail.timeline.factory
+
+import im.vector.app.features.home.room.detail.timeline.item.PollOptionViewState
+import im.vector.app.features.home.room.detail.timeline.item.PollResponseData
+import org.matrix.android.sdk.api.session.room.model.message.PollCreationInfo
+import javax.inject.Inject
+
+class PollOptionViewStateFactory @Inject constructor() {
+
+    fun createPollEndedOptions(pollCreationInfo: PollCreationInfo?, pollResponseData: PollResponseData?): List<PollOptionViewState.PollEnded> {
+        val winnerVoteCount = pollResponseData?.winnerVoteCount
+        return pollCreationInfo?.answers?.map { answer ->
+            val voteSummary = pollResponseData?.getVoteSummaryOfAnOption(answer.id ?: "")
+            PollOptionViewState.PollEnded(
+                    optionId = answer.id.orEmpty(),
+                    optionAnswer = answer.getBestAnswer().orEmpty(),
+                    voteCount = voteSummary?.total ?: 0,
+                    votePercentage = voteSummary?.percentage ?: 0.0,
+                    isWinner = winnerVoteCount != 0 && voteSummary?.total == winnerVoteCount
```

Comment:
Can we expect all the default values under normal conditions? Or will this break the UI?

Reply:
I am not sure to undertand your question. Could you add more details?

Reply:
Based on the code, I'd expect that the answer ID, vote count, vote percentage etc. could all be null. But I'm wondering if any of those were to be null, would that be an unexpected error state? And would the UI react gracefully if it saw the default fallback variables (i.e. would it make sense to the user)?

Reply:
When default values will be used because we do not have values in the data model, it will render a poll message with empty content. I guess we do not show error in these situations as this will rarely happen.